app/js/view/upload_view.js

```diff
@@ -0,0 +1,208 @@
+    this.submitButton.addEventListener('click', this.upload.bind(this));
+    this.createList();
+  }
+
+  showAlertDialog(msg) {
+    this.alertDialog.textContent = msg;
+    this.alertDialog.open();
+  }
+
+  isEligible(app) {
+    if (this.uploadBlacklist.indexOf(app.manifestURL) !== -1) {
+      return false;
+    }
+    if (app.manifest.role === 'theme' || app.manifest.role === 'addon') {
+      return true;
+    }
+    // Only non-gaia non-marketplace apps are eligible for hackerplace.
+    return (!app.installOrigin.endsWith('gaiamobile.org') &&
+            app.installOrigin !== 'https://marketplace.firefox.com');
+  }
+
+  createList() {
+    var req;
+    if (navigator.mozApps.mgmt) {
+      req = navigator.mozApps.mgmt.getAll();
+    } else {
+      req = navigator.mozApps.getInstalled();
+    }
+    req.onsuccess = () => {
+      var apps = req.result;
+      apps.forEach(app => {
+        if (this.isEligible(app)) {
+          var item = document.createElement('li');
+          var icon = this.getIconUrl(app.manifest, app.origin);
+          item.classList.add('item');
+          item.innerHTML = this.itemTemplate(app.manifest);
+          IconHelper.setImage(item.querySelector('.icon'), icon);
+          this.list.appendChild(item);
+          item.addEventListener('click', this.showForm.bind(this, app));
+        }
+      });
+    };
+    req.onerror = e => {
+      console.log('Unable to fetch installed apps.', e);
+    };
+  }
+
+  getIconUrl(manifest, origin) {
+    if (!manifest || !manifest.icons) {
+      return null;
+    }
+    var url;
+    for (var size in manifest.icons) {
+      url = manifest.icons[size];
+    }
+    if (url.startsWith('/')) {
+      url = origin + url;
+    }
+    return url;
+  }
+
+  showForm(app) {
+    this.nameInput.value =
+      app.manifest.developer && app.manifest.developer.name || '';
+    this.descriptionInput.value = app.manifest.description || '';
+    this.currentApp = app;
+    this.displayName.textContent = app.manifest.name;
+    IconHelper.setImage(this.displayIcon,
+      this.getIconUrl(app.manifest, app.origin));
+    this.el.classList.add('form');
+  }
+
+  hideForm(app) {
+    this.nameInput.value = '';
+    this.descriptionInput.value = '';
+    this.currentApp = null;
+    this.el.classList.remove('form');
+  }
+
+  show() {
+    this.el.classList.add('active');
+  }
+
+  hide() {
+    this.el.classList.remove('active');
+  }
+
+  upload() {
+    if (!this.currentApp) {
+      this.showAlertDialog('Error: current app not found');
+      this.hideForm();
+      return;
+    }
+
+    if (this.nameInput.value === '') {
+      this.showAlertDialog('You must fill out a name');
+      return;
+    }
+
+    this.currentApp.export().then((blob) => {
+      var name = encodeURIComponent(this.nameInput.value);
+      var description = encodeURIComponent(this.descriptionInput.value);
+      var url =
+        `http://104.236.138.217/upload?name=${name}&description=${description}`;
```

Comment:
Based on an IP whois, this looks like it's your VPS. Could we at least get a domain for it?